source/validations.ts

```diff
@@ -313,6 +313,46 @@ const validations = {
 		}
 	},
 
+	knownOptions(passedOptions?: Partial<Options>) {
+		if (!passedOptions) return // no options is a valid config
+
+		type KeysEnum<T> = { [P in keyof Required<T>]: true }
+		// we have to manually update this when adding new options, because we want it to work even for JS users. But KeysEnum makes TypeScript ensures it's correct at build time!
+		const optionsMap: KeysEnum<Options> = {
```

Comment:
I don't like that we have to repeat the entire list again (once in the types, once as part of the `Configuration`), but oh well.